File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        return dfs(nums,0,-1);
    }

    private int dfs(int[] nums,int i, int j)
    {
        if(i==nums.length) return 0;

        int len= dfs(nums,i+1,j);

        if(j==-1 || nums[j]<nums[i])
        {
            len=Math.max(1+dfs(nums,i+1,i),len);
        }
        return len;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        '''
            brute force :-
            two loops

            O(n^2)
        '''
        res = -10000000
        for i in range(len(nums)):
            Sum = nums[i]
            res = max(res,Sum)
            for j in range(i+1,len(nums)):
                
                Sum += nums[j]
                res = max(res,Sum)
        
        return res